### ISSUE TITLE
Tokens highlighted inside others

### ISSUE THREAD

**Issue by NathanFallet** [markusressel/KodeHighlighter#21]
**Describe the bug**
When a token is recognized in another token, it is highlighted: it should not!
For example, a class name or a keyword inside a comment or a string should not be highlighted with another color.

**To Reproduce**
1. Open the demo app embedded in the project
2. Scroll to a piece of code with tokens in comments (there are many examples: a `is` in the python example, or many in the ocaml example)

**Expected behavior**
In a comment, everything should be highlighted as a comment

**Screenshots**
![Screenshot_2022-03-30-18-30-44-526_de markusressel kodehighlighter demo_Original](https://user-images.githubusercontent.com/30439790/160886106-4e910be3-b06a-4f7c-9abf-e9ee06c4cf72.jpg)
![Screenshot_2022-03-30-18-31-17-536_de markusressel kodehighlighter demo_Original](https://user-images.githubusercontent.com/30439790/160886114-10e0f163-2137-45c1-b752-e53d7a172549.jpg)

**Smartphone (please complete the following information):**
 - Device: Xiaomi Redmi 9
 - OS: Android 11
 But the problem is not related to the device or the Android version

**Comment by markusressel** [comment]
True, I have seen this too, but haven't had the chance to fix it.

**Comment by NathanFallet** [comment]
I'm going to take a look at the library I use on iOS to see how parser is handling this.
